### PR TITLE
libasynql: Implement PM5 and pmmpthread 6.0 changes

### PR DESCRIPTION
**Pull request by alvin0319**

### CODE REVIEW

File: libasynql/src/poggit/libasynql/base/QueryRecvQueue.php [hunk 1]
```diff
@@ -24,33 +24,38 @@
 
 use poggit\libasynql\SqlError;
 use poggit\libasynql\SqlResult;
-use Threaded;
```
Comment (SOF3) [inline libasynql/src/poggit/libasynql/base/QueryRecvQueue.php:27]: Code style nit: use imports instead of FQN

**Review by SOF3 — APPROVED** [review]

File: libasynql/src/poggit/libasynql/base/SqlSlaveThread.php [hunk 2]
```diff
@@ -63,14 +60,16 @@ protected function __construct(SleeperNotifier $notifier, QuerySendQueue $buffer
 			$cl = Server::getInstance()->getPluginManager()->getPlugin("DEVirion")->getVirionClassLoader();
 			$this->setClassLoaders([Server::getInstance()->getLoader(), $cl]);
 		}
-		$this->start(PTHREADS_INHERIT_INI | PTHREADS_INHERIT_CONSTANTS);
+		$this->start(NativeThread::INHERIT_INI | NativeThread::INHERIT_CONSTANTS);
```
Comment (dktapps) [inline libasynql/src/poggit/libasynql/base/SqlSlaveThread.php]: I don't recommend using INHERIT_CONSTANTS. Unless you have global constants that are required, I recommend removing this.

File: libasynql/src/poggit/libasynql/base/SqlSlaveThread.php [hunk 3]
```diff
@@ -22,22 +22,19 @@
...
 use poggit\libasynql\SqlResult;
 use poggit\libasynql\SqlThread;
-use const PTHREADS_INHERIT_CONSTANTS;
-use const PTHREADS_INHERIT_INI;
-use function assert;
 
 abstract class SqlSlaveThread extends Thread implements SqlThread{
-	/** @var SleeperNotifier */
-	private $notifier;
+	private SleeperHandlerEntry $entry;
```
Comment (dktapps) [inline libasynql/src/poggit/libasynql/base/SqlSlaveThread.php]: The naming of this property seems a bit too generic

File: libasynql/src/poggit/libasynql/base/SqlThreadPool.php [hunk 4]
```diff
@@ -24,12 +24,12 @@
...
 use pocketmine\Server;
-use pocketmine\snooze\SleeperNotifier;
+use pocketmine\snooze\SleeperHandlerEntry;
 use poggit\libasynql\SqlThread;
 
 class SqlThreadPool implements SqlThread{
-	/** @var SleeperNotifier */
-	private $notifier;
+
+	private SleeperHandlerEntry $entry;
```
Comment (dktapps) [inline libasynql/src/poggit/libasynql/base/SqlThreadPool.php]: again, too generic

**Comment by SOF3** [comment]
I am not familiar with with new API changes, will leave this to dylan since there seems to be no semantic changes

**Review by SOF3 — CHANGES_REQUESTED** [review]
Please also update the pmmp dependency in composer.json

**Review by SOF3 — APPROVED** [review]

File: composer.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
     "name": "sof3/libasynql",
     "type": "library",
     "require": {
-        "pocketmine/pocketmine-mp": "^4.0.0"
+        "pocketmine/pocketmine-mp": "dev-major-next"
```
Comment (dktapps) [inline composer.json]: `^5.0.0-BETA4` would be better for this.

**Review by dktapps — APPROVED** [review]
The code looks fine. It could probably be optimised some, but I don't think that's a concern for this PR (libasynql was already not optimal before this PR anyway).

File: composer.json (hunk 5 above)
Comment (SOF3) [reply composer.json]: we're not tagging the pm5 branch though, why do we need to lock in to a version?

Comment (dktapps) [reply composer.json]: we're not locking in. The ^ allows any future version of 5.x to be used.

Comment (alvin0319) [reply composer.json]: does `^5.0.0` work for any beta version? Looks like `^5.0.0` would be better for future.

Comment (dktapps) [reply composer.json]: Not sure. You're welcome to try it.

Comment (alvin0319) [reply composer.json]: Looks like it doesn't work.

Comment (dktapps) [reply composer.json]: ^5.0.0-BETA4 will accept any 5.x.y release anyway, so this is fine.

**Pull request merged**